webapp/src/pages/createApplicationForm/components/Section.js

```diff
@@ -55,90 +69,206 @@ const Section = ({
         fr: ''
       },
       required: false
-    }]});
+    }
+    const updatedSections = sections.map(s => {
+      if (s.id === input.id) {
+        s = {...input, questions: [...qsts, qst]};
+      }
+      return s;
+    });
+    setInput({...input, questions: [...qsts, qst]});
+    setApplytransition(false);
+    setQuestionAnimation(false);
   }
 
   const handleQuestions = (inpt) => {
     setInput({...input, questions: inpt});
   }
 
+  const handleOkDelete = () => {
+    const updatedSections = sections.filter(s => s.id !== input.id);
+    setSection(updatedSections);
+  }
+
+  const handleConfirm = () => {
+    setIsModelVisible(!isModelVisible);
+  }
+
+  const handleMoveSUp = () => {
+    handleMove({
+      elements: sections,
+      index: sectionIndex,
+      setState: setSection,
+      u: true
+    });
+  }
+
+  const handleMoveSDown = () => {
+    handleMove({
+      elements: sections,
+      index: sectionIndex,
+      setState: setSection
+    });
+  }
+
+  const handleDragQ = (e) => {
+    drag(e, setDragId);
+  }
+
+  const handleDropQ = (e) => {
+    drop({
+      event: e,
+      elements: input.questions,
+      dragId,
+      setState: setInput,
+      section: input,
+      setAnimation: setQuestionAnimation,
+      setSection: setSection,
+      sections
+    });
+    setParentDropable(false);
+    setApplytransition(false);
+  }
+
+  const handleMouseDown = () => {
+    setParentDropable(true);
+  }
+
+  const handleMouseOut = () => {
+    setParentDropable(false);
+  }
+
   const numSections = sections.length;
   const index = sectionIndex + 1;
+  const isDisabled = sections.length === 1 ? true : false;
```

Comment:
Please update the name here to make it clearer what exactly is disabled (eg `isDeleteDisabled`)